src/lib/core/traverse.js

```diff
@@ -59,7 +59,10 @@ function traverse(schema, path, resolve, rootSchema) {
     const retval = utils.typecast(null, schema, () => schema.generate(rootSchema, path));
     const type = typeof retval;
 
-    if (type === schema.type || (type === 'number' && schema.type === 'integer') || (Array.isArray(retval) && schema.type === 'array')) {
+    if (type === schema.type
+       || (type === 'number' && schema.type === 'integer')
+       || (Array.isArray(retval) && schema.type === 'array')
+       || (retval === null || retval === undefined)) {
```

Comment:
I would allow nulls only if the type is null, not all nulls, e.g. `schema.type === 'null' && retval === null`

Also, why allow undefined values? Those are not expected per-se according json-schema, they're really needed?

Reply:
I can remove the undefined one - it is just to account for both since it is a little fuzzy on standards. It seems json stringify removes it if not specified? Anyway, will do.

The null check is tricky - because per json schema - if I can only have null value then. That is not my scenario - which is more "string or null". Looking more into this - there are two ways in JSON schema to do this - in the latest 7.0 it is to use an array of types: ```{ "type": ["string", "null"] }```. Earlier versions use `anyOf`.

Do you have suggestions on what to go by?

Reply:
Additionally - if I may point out - even the Array check is really not complete that way since array.items type is not verified. Do you want to address all these together?

Reply:
Fine, please remove the undefined one and let the other check as is, if it passes the tests with `make ci` in your machine I think i'll be fine then. I also think that `type:[a,b]` is fine and totally supported by this library, so if we can stick on _type checks_ it'll be better, thank you!

Reply:
If tests are fine, I believe we can omit the type check for nulls — also, please try adding a new spec/schema for testing this use-case, that would help us to understand the issue more accurately.

Reply:
I will add the testcase. Just pushed the type check for null. Can you review?